Add support for >256 buffers and bugfix

MX682X opened the pull request:
fix and improvement as mentioned in #839

SpenceKonde commented:
I'm continuing to be confused with the changes here and in DxCore relating to Wire library, because I thought there were additional changes from DxCore waiting to be merged - the development effort is outrunning my ability to keep up with it and know what needs to be merged into where.  Can you let me know what else, if anything, needs to go into mTC and what needs to be ported to DxC for the next release wrt Wire?

MX682X commented:
Alright, let's see. My PR in DxCore had the main target of adding an example for Dual Mode. But since the github version was not compiling, I decided to change it enough to make it compile. Also, I did some housekeeping in the Wire library itself. Later I realized, I should keep PRs as simple as possible...
Anyway, as far as I know, I also implemented the changes from DxCore to mTC aswell, except the Dual Mode example itself, since the tinies do not support it. Basically, you just have to move the files in the src folder to DxC.

Maybe also useful, a summary of the changes, in chronolgical order:
DxC: added readBytes() function (as known from Serial), removed BusErr enabling and checks. (Due to transmission failure after sleep), made the TWI_checkPins check for scl - 1 == sda (like in Serial)
[PR closed, but those are in mTC]

mTC: add popSleep and pushSleep functions to fix sleep handling, also optimzations to keep the size low. Like: convincing Compiler to use Y-Register in Slave ISR without writing everything in assembly. But this required merging the subfunctions.

mTC: while optimizing, I made a mistake, and this was fixed in this PR.
